src/Validator/Constraints/InfiniteLoopValidator.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Setono\SyliusRedirectPlugin\Validator\Constraints;
+
+use Setono\SyliusRedirectPlugin\Model\RedirectInterface;
+use Setono\SyliusRedirectPlugin\Repository\RedirectRepositoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\Form\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+final class InfiniteLoopValidator extends ConstraintValidator
+{
+    /** @var RepositoryInterface|RedirectRepositoryInterface */
```

Comment:
The `RedirectRepositoryInterface` extends `RepositoryInterface` so you only need the `RedirectRepositoryInterface` here